x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java

```diff
@@ -53,24 +59,28 @@ public static class Request extends ActionRequest implements AnalyticsEvent.Cont
 
         private final XContentType xContentType;
 
-        public Request(String eventCollectionName, String eventType, boolean debug, XContentType xContentType, BytesReference payload) {
-            this(eventCollectionName, eventType, debug, System.currentTimeMillis(), xContentType, payload);
-        }
+        private final Map<String, List<String>> headers;
+
+        private final InetAddress clientAddress;
 
-        public Request(
+        private Request(
             String eventCollectionName,
             String eventType,
-            boolean debug,
             long eventTime,
             XContentType xContentType,
-            BytesReference payload
+            BytesReference payload,
+            boolean debug,
+            @Nullable Map<String, List<String>> headers,
+            @Nullable InetAddress clientAddress
```

Comment:
I find it painful to manipulate `InetAddress` and would prefer to have string instead.
It will simplify a lot the code (especially serialization / unserialization).

Reply:
I prefer to do strong typing where possible, and ensure it's a proper InetAddress - but I'll leave it up to you